### ISSUE TITLE
ecs_service.py: "network_configuration" for "awsvpc" is missing "assign_public_ip"

### ISSUE THREAD

**Issue by tomerr-tradair** [ansible/ansible#41291]
##### SUMMARY
Cannot assign public IP to an ECS container running with awsvpc network_configuration mode (also affects fargate probably).
https://www.terraform.io/docs/providers/aws/r/ecs_service.html#assign_public_ip
The subnet and security group options are OK but the 3rd option is missing from the code.

##### ISSUE TYPE
 - Bug Report

##### COMPONENT NAME
amazon/ecs_service

##### ANSIBLE VERSION
latest development

##### STEPS TO REPRODUCE
- name: Create new / Update existing ECS service
  ecs_service:
    state: present
    name: "service-{{ hostName }}"
    cluster: "{{ ecsClusterName[region] }}"
    task_definition: "marc-task-{{ hostName }}"
    desired_count: 1
    region: "{{ region }}"
    deployment_configuration:
      minimum_healthy_percent: 50
      maximum_percent: 200
    network_configuration:
      subnets: "{{ subnets[region] }}"
      security_groups:
      - SG-INT-ALL
      assign_public_ip: true

##### EXPECTED RESULTS

assignPublicIp should be ENABLED

##### ACTUAL RESULTS
  launchType: EC2
02:11:29       networkConfiguration:
02:11:29         awsvpcConfiguration:
02:11:29           assignPublicIp: DISABLED
02:11:29           securityGroups:
02:11:29           - sg-a89fc1d6
02:11:29           subnets:
02:11:29           - subnet-6c9c4d42
02:11:29       pendingCount: 1
02:11:29       runningCount: 0
02:11:29       status: ACTIVE

**Comment by ansibot** [comment]
Files identified in the description:
* [lib/ansible/modules/cloud/amazon/ecs_service.py](https://github.com/ansible/ansible/blob/devel/lib/ansible/modules/cloud/amazon/ecs_service.py)

If these files are inaccurate, please update the `component name` section of the description or use the `!component` bot command.

[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by ansibot** [comment]
cc @java1guy @kaczynskid @willthames @zacblazic
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by willthames** [comment]
Please surround yaml with triple backticks, or the ansible task is very hard to read!

**Comment by willthames** [comment]
I remember this at the time. 

It looks like the awsvpc configuration was added in botocore 1.7.44, but assignPublicIp was only added with 1.8.4, a couple of weeks later. I did struggle to get assign_public_ip to work, probably because I was on a botocore version somewhere between the two.

Any implementation should take care to ensure that the module still works for people with 1.7.44 <= botocore < 1.8.4, with a friendly error message if they pass assign_public_ip but don't yet have 1.8.4, and the test suite would need to validate this use case.

needs_contributor
